hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/store/driver/TestStateStoreDriverBase.java

```diff
@@ -574,6 +580,38 @@ private static Map<String, Class<?>> getFields(BaseRecord record) {
     return getters;
   }
 
+  public long getMountTableCacheLoadSamples(StateStoreDriver driver) throws IOException {
+    final MutableRate mountTableCache = getMountTableCache(driver);
+    return mountTableCache.lastStat().numSamples();
+  }
+
+  private static MutableRate getMountTableCache(StateStoreDriver driver) throws IOException {
+    StateStoreMetrics metrics = stateStore.getMetrics();
+    final Query<MountTable> query = new Query<>(MountTable.newInstance());
+    driver.getMultiple(MountTable.class, query);
+    final Map<String, MutableRate> cacheLoadMetrics = metrics.getCacheLoadMetrics();
+    final MutableRate mountTableCache = cacheLoadMetrics.get("CacheMountTableLoad");
+    assertNotNull("CacheMountTableLoad should be present in the state store metrics",
+        mountTableCache);
+    return mountTableCache;
+  }
+
+  public void testCacheLoadMetrics(StateStoreDriver driver, long numRefresh)
+      throws IOException, IllegalArgumentException {
+    final MutableRate mountTableCache = getMountTableCache(driver);
+    // CacheMountTableLoadNumOps
+    final long mountTableCacheLoadNumOps = getMountTableCacheLoadSamples(driver);
+    assertEquals("Num of samples collected should match", numRefresh, mountTableCacheLoadNumOps);
+    // CacheMountTableLoadAvgTime ms
+    final double mountTableCacheLoadAvgTimeMs = mountTableCache.lastStat().mean();
+    // 2 seconds is a high enough value for the test, hence we expect mount table cache
+    // with very few entries to be loaded by this time duration, hence not have this test result
+    // show flaky behavior.
+    assertTrue(
+        "Mean time duration for cache load is expected to be less than 2000 ms. Actual value: "
+            + mountTableCacheLoadAvgTimeMs, mountTableCacheLoadAvgTimeMs < 2000d);
+  }
```

Comment:
Isn't just asserting this value ain't zero enough?

Reply:
That also seems good, either way I think it's fine. The values I have seen have been less than 5 ms so far hence crossing 2000 ms seems like really big deal, even for Jenkins CI, WDYT? 
But yeah on the other hand, `mountTableCacheLoadAvgTimeMs > 0d` might also be good enough.

Shall we do `0 < mountTableCacheLoadAvgTimeMs < 2000`?

Reply:
If it more than 0, your metrics is getting captured and propogated, we are fine with that

if I debug this test, it would fail only here, moreover why introduce a scope of a flaky test which can be dodged

Reply:
Cool, that's fine. I am only worried about fast Jenkins env where just in case if the cache loading happens in nanoseconds, perhaps even milliseconds might be 0?

Reply:
Even if it goes to say 10K, it has nothing to do with your code, your code still captured the value and we would be happy with that.

Now if it is 0, it means it didn't capture the value, that is where we need to worried.

if there are chances it can stay at 0 :

- If only in test: May be have the default value as -1 and we can assert it isn't -1 and we should be sorted, if it is -1 then things didin't work the way we wanted.
- If it can be in prod as well: then we might have to think of lowering the unit or have dynamic units or so, because having these values as 0 in a prod cluster would be just confusing? mostly would create doubts like as if the metrics isn't getting captured.

Reply:
Oh for prod, I think we should be better with ms itself because even with few entries I see at least some value in double for sure and that is always > 0 so I am sure we should be good there.
For test also, I just did more than 10 runs for each state store impl tests: fs, dfs, zk. And we definitely have values greater than 0. I think you are right that this should not be 0, let's go with 0 for now and if Jenkins QA complains, I would be happy to investigate further and we might apply comparison with -1 too. For now, 0 sounds good.